c7n/resources/appmesh.py

```diff
@@ -324,3 +327,125 @@ class resource_type(TypeInfo):
             'virtualNodes',
             None,
         )
+
+
+@AppmeshMesh.filter_registry.register('virtual-service')
+class VirtualService(ListItemFilter):
+
+    schema = type_schema(
+        'virtual-service',
+        attrs={'$ref': '#/definitions/filters_common/list_item_attrs'},
+        count={'type': 'number'},
+        count_op={'$ref': '#/definitions/filters_common/comparison_operators'}
+    )
+
+    annotation_key = "virtualServices"
+    permissions = ('appmesh:DescribeVirtualService',)
+
+    def __init__(self, data, manager=None):
+        super().__init__(data, manager)
+        self.data['key'] = self.annotation_key
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('appmesh')
+        for mesh in resources:
+            response = self.manager.retry(client.list_virtual_services,
+                                          meshName=mesh["meshName"])
+
+            mesh[self.annotation_key] = response.get(
+                'virtualServices', [])
+
+        return super().process(resources, event)
+
+
+@AppmeshMesh.filter_registry.register('virtual-router')

```

Comment:
same with this one re s/virtual/appmesh

Reply:
This is fixed to change it to appmesh-router. Please check